Engine/DataFeeds/Enumerators/Factories/FuturesChainUniverseSubscriptionEnumeratorFactory.cs

```diff
@@ -0,0 +1,123 @@
+        /// <returns>An enumerator reading the subscription request</returns>
+        public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IDataFileProvider dataFileProvider)
+        {
+            if (_isLiveMode)
+            {
+                var localTime = request.StartTimeUtc.ConvertFromUtc(request.Configuration.ExchangeTimeZone);
+                
+                // loading the list of futures contracts and converting them into zip entries
+                var symbols = _symbolUniverse.LookupSymbols(request.Security.Symbol.Underlying.Value, request.Security.Type);
+                var zipEntries = symbols.Select(x => new ZipEntryName { Time = localTime, Symbol = x } as BaseData).ToList();
+
+                var underlyingEnumerator = new TradeBarBuilderEnumerator(request.Configuration.Increment, request.Security.Exchange.TimeZone, _timeProvider);
+                underlyingEnumerator.ProcessData(new Tick { Value = 0 });
+
+                // configuring the enumerator
+                var subscriptionConfiguration = GetSubscriptionConfigurations(request).First();
+                var subscriptionRequest = new SubscriptionRequest(request, configuration: subscriptionConfiguration);
+                var configuredEnumerator = _enumeratorConfigurator(subscriptionRequest, underlyingEnumerator);
+
+                return new DataQueueFuturesChainUniverseDataCollectionEnumerator(request.Security.Symbol, configuredEnumerator, zipEntries);
+            }
+            else
+            {
+                var factory = new BaseDataSubscriptionEnumeratorFactory();
+
+                var enumerators = GetSubscriptionConfigurations(request)
+                    .Select(c => new SubscriptionRequest(request, configuration: c))
+                    .Select(sr => _enumeratorConfigurator(request, factory.CreateEnumerator(sr, dataFileProvider)));
+
+                var sync = new SynchronizingEnumerator(enumerators);
+                return new FuturesChainUniverseDataCollectionAggregatorEnumerator(sync, request.Security.Symbol);
+            }
+        }
+
+        private IEnumerable<SubscriptionDataConfig> GetSubscriptionConfigurations(SubscriptionRequest request)
+        {
+            // canonical also needs underlying price data
+            var config = request.Configuration;
+            var underlying = config.Symbol.Underlying;
+            var resolution = config.Resolution;
+
+            var configurations = new List<SubscriptionDataConfig>
+            {
+                // add underlying trade data
+                new SubscriptionDataConfig(config, resolution: resolution, fillForward: true, symbol: underlying, objectType: typeof (TradeBar), tickType: TickType.Trade),
+            };
+
+            if (!_isLiveMode)
+            {
+                // rewrite the primary to be non-tick and fill forward
+                configurations.Add(new SubscriptionDataConfig(config, resolution: resolution, fillForward: true));
```

Comment:
The comment says non-tick, but I don't see any code preventing `resolution == Resolution.Tick`